lib/storage/__mocks__/index.js

```diff
@@ -0,0 +1,3 @@
+import WebStorage from '../WebStorage';
```

Comment:
I see we have have `App/__mocks__`. Should we be consistent with where we place the mocks?

Reply:
I can move this to `{root}/__mocks__/lib/storage/index.js` as long as it represents the correct folder structure the mock will be used. 

Keep in mind Jest suggest to put the `__mocks__` directory adjacent to the module being mocked so that's how everybody's been taught. In this case we're mocking our own `index.js` file so I put it there
https://jestjs.io/docs/manual-mocks#mocking-user-modules

In App we're only mocking external dependencies coming from `node_modules` and so `__mocks__` is at the root

I supposed you would ask "why do we mock something that's not external?". It's because we're testing functionality that doesn't depend on the actual storage implementation, we're asserting that certain calls are made at the expected times with the expected data. It was the same with AsyncStorage - we used the mock of AsyncStorage to verify we're calling `setItem` with the expected parameters. Here we're passing `WebStorage` which would internally use the mock of AsyncStorage, but would also execute the custom logic we've added (e.g. stringify JSONs)